### PR TITLE
localization updates

### PR DESCRIPTION
**Pull request by lts20050703**
Closes: #2155 

Upstream: https://github.com/discord/discord-api-docs/commit/2f854d3fd675a9953d1bb8d6858a03c8a9642337

Types: DiscordApplicationCommmand and DiscordApplicationCommandOption: Update documentation

Types: DiscordActivityButton: Add documentation link

Helper: upsertApplicationCommand: All fields are optional, but any fields provided will entirely overwrite the existing values of those fields.

### CODE REVIEW

File: types/discord.ts [hunk 1]
```diff
@@ -924,8 +924,7 @@ export interface DiscordActivitySecrets {
   match?: string;
 }
 
-// https://github.com/discord/discord-api-docs/pull/2219
-// TODO: add documentation link
+/** https://discord.com/developers/docs/topics/gateway#activity-object-activity-buttons */
```
Comment (lts20050703) [inline types/discord.ts:927]: Tell me if I should move this to another PR

File: types/shared.ts [hunk 2]
```diff
@@ -1285,53 +1285,6 @@ export type Camelize<T> = {
...
-//   & {
-//     [K in Exclude<keyof T, KeysWithUndefined<T>>]: T[K] extends object ? Object extends Pick<T[K], keyof T[K]> ? T[K]
-//     : T[K] extends Collection<any, any> ? T[K]
-//     : T[K] extends any[] ? T[K]
-//     : Optionalize<T[K]>
-//       : T[K];
-//   }
-// )
-//   : T;
-
```
Comment (lts20050703) [inline types/shared.ts:1334]: tell me if I should move this to another PR

File: helpers/interactions/commands/createApplicationCommand.ts [hunk 3]
```diff
@@ -89,7 +89,7 @@ export interface CreateContextApplicationCommand {
 }
 
 export function isContextApplicationCommand(
-  cmd: CreateContextApplicationCommand | CreateApplicationCommand,
-): cmd is CreateContextApplicationCommand {
+  cmd: Partial<CreateContextApplicationCommand> | Partial<CreateApplicationCommand>,
+): cmd is Partial<CreateContextApplicationCommand> {
```
Comment (itohatweb) [inline helpers/interactions/commands/createApplicationCommand.ts:93]: I do not like this change

File: types/discord.ts (hunk 1 above)
Comment (itohatweb) [reply types/discord.ts:927]: I think this is the answer 🙃

File: types/shared.ts (hunk 2 above)
Comment (itohatweb) [reply types/shared.ts:1334]: I think this is the answer 🙃

File: types/discord.ts [hunk 4]
```diff
@@ -1645,7 +1644,7 @@ export interface DiscordApplicationCommand {
   options?: DiscordApplicationCommandOption[];
   /** Whether the command is enabled by default when the app is added to a guild */
   default_permission?: boolean;
-  /** The type of command. By default this is a application command(ChatInput). */
+  /** The type of command. By default this is a application command (ApplicationCommandTypes.ChatInput). */
```
Comment (itohatweb) [inline types/discord.ts:1647]: I do not think we need this change in this case.

File: types/discord.ts [hunk 5]
```diff
@@ -1633,7 +1632,7 @@ export interface DiscordApplicationCommand {
   application_id: string;
   /** Guild id of the command, if not global */
   guild_id?: string;
-  /** 1-32 character name matching */
+  /** `ApplicationCommandTypes.ChatInput` command names must match the following regex `^[-_\p{L}\p{N}\p{sc=Deva}\p{sc=Thai}]{1,32}$` with the unicode flag set. If there is a lowercase variant of any letters used, you must use those. Characters with no lowercase variants and/or uncased letters are still allowed. `ApplicationCommandTypes.User` and `ApplicationCommandTypes.Message` commands may be mixed case and can include spaces. */
```
Comment (itohatweb) [inline types/discord.ts:1635]: There are more interfaces affected by this change. Please update all of them. 
Also update the permission plugin accordingly.

**Review by itohatweb — CHANGES_REQUESTED** [review]

File: helpers/interactions/commands/createApplicationCommand.ts (hunk 3 above)
Comment (lts20050703) [reply helpers/interactions/commands/createApplicationCommand.ts:93]: @itohatweb Why? The Docs stated that **All fields are optional**, but any fields provided will entirely overwrite the existing values of those fields.

Comment (itohatweb) [reply helpers/interactions/commands/createApplicationCommand.ts:93]: I feel like it's more the name of the function itself which is wrong here.

File: types/discord.ts (hunk 5 above)
Comment (lts20050703) [reply types/discord.ts:1636]: updated regex

File: helpers/interactions/commands/createApplicationCommand.ts (hunk 3 above)
Comment (lts20050703) [reply helpers/interactions/commands/createApplicationCommand.ts:93]: How should I fix this?

Comment (lts20050703) [reply helpers/interactions/commands/createApplicationCommand.ts:93]: resolved in discord. 
> just ignore
> let this be
> what u have rn
> let this be how it is now
> yes
> like i said
> leave this

**Review by Skillz4Killz — APPROVED** [review]
